### PR TITLE
Add listener for km

### PR DESCRIPTION
**Pull request by attiks**

### CODE REVIEW

**Comment by attiks** [comment]
See https://github.com/UN-OCHA/docstore-site/pull/119

**Comment by attiks** [comment]
@orakili code seems to be working fine, if a term is updated on the document store, the webhook is fired and the code scans for referenced entities, the only problem is the language. Code at html/modules/contrib/search_api/src/Plugin/search_api/datasource/ContentEntity.php assumes the entity has a language, but this might be a problem in External Entities. It falls back to all the site languages, but the indexing is done using `und`

```
        foreach ($entity_ids as $entity_id) {
          foreach ($this->getLanguages() as $language) {
            $ids_to_reindex["$entity_id:{$language->getId()}"] = 1;
          }
        }
```
if changed to the following, it will index the referenced entities as well
```
        foreach ($entity_ids as $entity_id) {
          $ids_to_reindex["$entity_id:und"] = 1;
          foreach ($this->getLanguages() as $language) {
            $ids_to_reindex["$entity_id:{$language->getId()}"] = 1;
          }
        }
```

**Comment by attiks** [comment]
Proper solution is to implement our own `ContentEntity` class so we can control the language.

File: html/modules/custom/ocha_docstore_files/src/Controller/WebhookController.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        break;
+
+      case 'assessment_document':
+        $index = Index::load('assessments');
+        $datasource_id = 'entity:assessment';
+        $external_entity_type = 'assessment_document';
+        break;
+
+      case 'term':
+        $datasource_id = 'organization';
```
Comment (orakili) [inline html/modules/custom/ocha_docstore_files/src/Controller/WebhookController.php:75]: That looks strange, also no index info.

**Review by orakili — COMMENTED** [review]
I think something is missing for the terms, also the events may change in https://github.com/UN-OCHA/docstore-site/pull/119.

File: html/modules/custom/ocha_docstore_files/src/Controller/WebhookController.php (hunk 1 above)
Comment (attiks) [reply html/modules/custom/ocha_docstore_files/src/Controller/WebhookController.php:75]: Code needs to be adapted so we can easily handle deletes

**Comment by orakili** [comment]
Docstore PR with new events: https://github.com/UN-OCHA/docstore-site/pull/124

**Comment by orakili** [comment]
@attiks PR that adds a search API datasource plugin: https://github.com/UN-OCHA/assessmentregistry8-site/pull/211

It contains notably a change similar to what you mentioned in https://github.com/UN-OCHA/assessmentregistry8-site/pull/198#issuecomment-804140043

**Pull request merged**